system/security/shared/seclib.hpp

```diff
@@ -198,10 +217,60 @@ enum authStatus : int
     AS_PASSWORD_VALID_BUT_EXPIRED = 5//user entered valid password, but authentication failed because it is expired
 };
 
+enum SecUserFeatureIndex : unsigned
+{
+    SUF_GetName,
+    SUF_SetName,
+    SUF_GetFullname,
+    SUF_SetFullName,
+    SUF_GetFirstName,
+    SUF_SetFirstName,
+    SUF_GetLastName,
+    SUF_SetLastName,
+    SUF_GetEmployeeID,
+    SUF_SetEmployeeID,
+    SUF_GetEmployeeNumber,
+    SUF_SetEmployeeNumber,
+    SUF_GetDistinguishedName,
+    SUF_SetDistinguishedName,
+    SUF_GetRealm,
+    SUF_SetRealm,
+    SUF_GetFqdn,
+    SUF_SetFqdn,
+    SUF_GetPeer,
+    SUF_SetPeer,
+    SUF_GetStatus,
+    SUF_SetStatus,
+    SUF_GetAuthenticateStatus,
+    SUF_SetAuthenticateStatus,
+    SUF_Credentials,
+    SUF_GetUserID,
+    SUF_CopyTo,
+    SUF_GetPasswordExpiration,
+    SUF_SetPasswordExpiration,
+    SUF_GetPasswordDaysRemaining,
+    SUF_GetProperty,
+    SUF_SetProperty,
+    SUF_GetPropertyInt,
+    SUF_SetPropertyInt,
+    SUF_GetPropertyIterator,
+    SUF_Clone,
+    SUF_GetDataElement,
+    SUF_GetDataElements,
+
+    SUF_NUMBER_OF_INDICES
+};
+
 class CDateTime;
 interface IPropertyIterator;
 interface ISecUser : extends IInterface
 {
+    virtual uint64_t queryFeatures(SecurityFeatureSupportLevel level) const = 0;
+    inline bool checkFeatures(SecurityFeatureSupportLevel level, uint64_t mask) const { return ((queryFeatures(level) & mask) == mask); }
+    inline bool checkUnsafeFeatures(uint64_t mask) const { return checkFeatures(SFSL_Unsafe, mask); }
```

Comment:
Can we improve the variable name "mask" to some thing like "SUFmask" or something more meaningful to a plugin developer looking at this for the first time?